x-pack/plugins/xpack_main/server/lib/feature_registry/feature_registry.ts

```diff
@@ -130,6 +131,10 @@ function validateFeature(feature: Feature) {
   const { app = [], management = {}, catalogue = [] } = feature;
 
   Object.entries(feature.privileges).forEach(([privilegeId, privilegeDefinition]) => {
+    if (!privilegeDefinition) {
```

Comment:
That's a good point, throwing here makes sense. There isn't a legitimate use-case for this, it's just allowed by TS because of the optional properties. I will make this change.